### PR TITLE
Decouple expression languages from core infrastructure

### PR DESCRIPTION
**Pull request by wetneb**
Migrates the registration of expression languages provided by default to the controller.js, and removes references to GREL from functionally independent areas.

This is a first step towards having GREL in its own Maven module.

### CODE REVIEW

**Comment by wetneb** [comment]
@tfmorris any interest in this? I'd like to merge it to be able to start the Maven modularization PR.

File: main/src/com/google/refine/LookupCacheManager.java [hunk 1]
```diff
@@ -50,6 +49,8 @@
  */
 public class LookupCacheManager {
 
+    public static final String INDEX_COLUMN_NAME = "_OpenRefine_Index_Column_Name_";
```
Comment (tfmorris) [inline main/src/com/google/refine/LookupCacheManager.java:52]: This seems unrelated. Is it intended to be included here? The change itself seems fine, but seems odd to have here.

File: main/tests/server/src/com/google/refine/operations/column/ColumnAdditionByFetchingURLsOperationTests.java [hunk 2]
```diff
@@ -101,9 +101,8 @@ public class ColumnAdditionByFetchingURLsOperationTests extends RefineTest {
             "    \"status\" : \"pending\"\n" +
             " }";
 
-    @Override
     @BeforeTest
-    public void init() {
+    public void initOperation() {
```
Comment (tfmorris) [inline main/tests/server/src/com/google/refine/operations/column/ColumnAdditionByFetchingURLsOperationTests.java:105]: This seems unrelated

File: main/src/com/google/refine/model/recon/ReconciledDataExtensionJob.java [hunk 3]
```diff
@@ -250,11 +251,13 @@ protected ReconciledDataExtensionJob.DataExtension collectResult(
...
                     storeCell(rows, rowindex, colindex, v);
                 } else if (val.has("date")) {
-                    ToDate td = new ToDate();
-                    String[] args = new String[1];
-                    args[0] = val.get("date").asText();
-                    Object v = td.call(null, args);
-                    storeCell(rows, rowindex, colindex, v);
+                    Object date;
+                    try {
+                        date = CalendarParser.parseAsOffsetDateTime(val.get("date").asText());
```
Comment (tfmorris) [inline main/src/com/google/refine/model/recon/ReconciledDataExtensionJob.java:256]: This also seems unrelated, but even if it gets moved to another PR, I don't think we should be introducing direct calls to CalendarParser which we should be targeting to get rid of. Who's date format is this? If it's something like an ISO date, we should just use the Java classes directly. If it's a human date, it should probably stay wrapped in `ToDate()`

**Review by tfmorris — CHANGES_REQUESTED** [review]
This seems like 3 or 4 unrelated sets of changes lumped together, but perhaps I'm not fully understanding what's going on. The core parser changes seem fine (with the addition of the `since = ` on the deprecation annotation).

File: main/src/com/google/refine/LookupCacheManager.java (hunk 1 above)
Comment (wetneb) [reply main/src/com/google/refine/LookupCacheManager.java:52]: It looks like I should have explained the whole thing better, sorry. This change is intentional. Having this constant declared in `Cross` constitutes a dependency from `LookupCacheManager` (a core class) to `Cross` (a GREL function), which will prevent the separation of the GREL implementation as a separate Maven module which would depend on a core module.

File: main/tests/server/src/com/google/refine/operations/column/ColumnAdditionByFetchingURLsOperationTests.java (hunk 2 above)
Comment (wetneb) [reply main/tests/server/src/com/google/refine/operations/column/ColumnAdditionByFetchingURLsOperationTests.java:105]: If I remember correctly, it's a necessary change for the tests to pass after adding the GREL registration to `RefineTest` instead of being done statically (otherwise, this `init` method overrides that of `RefineTest` which doesn't get executed for this test).

File: main/src/com/google/refine/model/recon/ReconciledDataExtensionJob.java (hunk 3 above)
Comment (wetneb) [reply main/src/com/google/refine/model/recon/ReconciledDataExtensionJob.java:256]: The motivation is still the same: removing a dependency from a core class to a GREL function. Using `CalendarParser` means that we support almost the same sort of date formats as before, which could be useful for services which rely on that.
But that being said now that the specs mandate ISO dates (although only in the JSON schemas so far, we should fix that) it should probably be safe to switch to supporting ISO dates only.

Comment (tfmorris) [reply main/src/com/google/refine/model/recon/ReconciledDataExtensionJob.java:256]: Sorry for the delay in circling back on this. I think using Java classes to support ISO dates (only) would be a better approach.

**Review by tfmorris — APPROVED** [review]
Thanks. LGTM
